reqs/tests/link_agencies_tests.py

```diff
@@ -0,0 +1,123 @@
+        'agencyName': 'Department of Defense',
+        'agencyType': '',
+        'CFO_Act': '1',
+        'CIO_Council': None,
+    })
+    synch.sync_row({
+        'agencyAbbreviation': "DHS",
+        'agencyCode': '100',
+        'agencyName': 'Department of Homeland Security',
+        'agencyType': '',
+        'CFO_Act': '1',
+        'CIO_Council': None,
+    })
+    synch.sync_row({
+        'agencyAbbreviation': "GSA",
+        'agencyCode': '101',
+        'agencyName': 'General Services Administration',
+        'agencyType': '',
+        'CFO_Act': '1',
+        'CIO_Council': None,
+    })
+    synch.sync_row({
+        'agencyAbbreviation': "Justice",
+        'agencyCode': '102',
+        'agencyName': 'Department of Justice',
+        'agencyType': '',
+        'CFO_Act': '1',
+        'CIO_Council': None,
+    })
+    synch.sync_row({
+        'agencyAbbreviation': "FOO",
+        'agencyCode': '112',
+        'agencyName': 'Department of Foo',
+        'agencyType': '',
+        'CFO_Act': None,
+        'CIO_Council': None,
+    })
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize("impacted_entity,agencies,agency_groups", [
+    (("",), (set(),), (set(),)),
+    (("All agencies, Defense",), ({"DOD"},), ({"all-agencies"},)),
+    (("dhs, Defense",), ({"DHS", "DOD"},), (set(),)),
+    (("CFO, Defense",), ({"DOD"},), ({"cfo-act"},)),
+    (("CFO; Defense; all agencies",), ({"DOD"},),
+     ({"all-agencies", "cfo-act"},)),
+    (("All agencies", "DOD"), (set(), {"DOD"}), ({"all-agencies"}, set())),
+    (("CFO\nDHS\nGSA\nDOJ\nDOD",), ({"DHS", "DOD", "Justice", "GSA"},),
+     ({"cfo-act"},)),
+    (("All agencies\nDHS\nGSA\nDOJ\nDOD",),
+     ({"DHS", "DOD", "Justice", "GSA"},), ({"all-agencies"},)),
+    (("DHS\nGSA\nDOJ\nDOD",), ({"DHS", "DOD", "Justice", "GSA"},),
+     (set(),)),
+    (("Federal CFO Council, DHS\nGSA\nDOJ\nDOD",),
+     ({"DHS", "DOD", "Justice", "GSA"},), (set(),)),
+    (("All agencies except micro agencies\nDHS\nGSA\nDOJ\nDOD",),
+     ({"DHS", "DOD", "Justice", "GSA"},), (set(),)),
+    (("All agencies except Paperwork\nDHS\nGSA\nDOJ\nDOD",),
+     ({"DHS", "DOD", "Justice", "GSA"},), (set(),)),
+    (("department of homeland\ngeneral services\nJustice\nDefense",),
+     ({"DHS", "DOD", "Justice", "GSA"},), (set(),)),
+], ids=repr)
+def test_linking(populate_agencies, impacted_entity, agencies,
+                 agency_groups):
+    reqs = [mommy.make(Requirement, impacted_entity=_) for _ in
+            impacted_entity]
+    link_agencies.Command().handle()
+    for i, req in enumerate(reqs):
+        req_agencies = {_.abbr for _ in req.agencies.all()}
+        assert req_agencies == agencies[i]
+        req_groups = {_.slug for _ in req.agency_groups.all()}
+        assert req_groups == agency_groups[i]
+
+
+@pytest.mark.django_db
+def test_link_requirements(populate_agencies):
+    dod_req = mommy.make(Requirement, impacted_entity="DOD")
+    dod_agency = Agency.objects.get(abbr="DOD")
```

Comment:
Here's a use case where you may want to yield something in the `populate_agencies` fixture. Instead of re-querying, you may provide an interface like `populate_agencies.dod` and `populate_agencies.all_agencies`